### PR TITLE
[tests] Add disk size check util and mock

### PR DESCRIPTION
**Pull request by townsend2010**
This fixes the reliance on the amount of disk space available during tests which can be non-deterministic.

### CODE REVIEW

File: tests/test_daemon.cpp [hunk 1]
```diff
@@ -83,6 +86,8 @@ void PrintTo(const YAML::Node& node, ::std::ostream* os)
 
 namespace
 {
+const qint64 default_total_bytes{16106127360};
```
Comment (ricab) [inline tests/test_daemon.cpp:89]:
```suggestion
const qint64 default_total_bytes{16'106'127'360}; // 16G
```

**Review by ricab — APPROVED** [review]
Thanks for this, it works nicely :) 

We may have to mock the function at a larger scope if we ever increase the default disk, but I guess that would be easy. I have a minor suggestion inline, but approving already.

bors delegate+

File: tests/test_daemon.cpp (hunk 1 above)
Comment (townsend2010) [reply tests/test_daemon.cpp:89]: Oh, cool, I didn't know one could add `'` to make the number more readable.

**Comment by townsend2010** [comment]
> We may have to mock the function at a larger scope if we ever increase the default disk, but I guess that would be easy.

Yeah, it would be real easy to do that by adding a default mock return in the `MockDaemon` which would affect every test that didn't `REPLACE` the function.

**Comment by townsend2010** [comment]
bors r=ricab

File: tests/test_daemon.cpp (hunk 1 above)
Comment (ricab) [reply tests/test_daemon.cpp]: Yeah introduced in C++14

**Pull request merged**